src/CoreArbiterServer.cc

```diff
@@ -1067,8 +1067,12 @@ CoreArbiterServer::distributeCores()
                     if (!sendData(thread->socket, &core->id, sizeof(int),
                                   "Error sending core ID to thread " +
                                         std::to_string(thread->id))) {
-                        exit(-1);
-                        return;
+                        // TODO(Qian): is it a good solution to ignore error?
+                        // The client side did not close TCP connection 
+                        // in a normal way, which caused the failure
+                        LOG(WARNING, "Pending to clean up state for thread %s",
```

Comment:
I'm not sure this extra log message is adding anything, unless the original error on line 1068 is not printing for some reason?